### PR TITLE
Add location priority

### PR DESCRIPTION
**Pull request by richfitz**
This PR adds support for the idea of location priority (or "trust" as I've called it in the ticket).

The idea here generalises a bit orderly's concept of letting dependencies from drafts override "more trustable" dependencies from the archive (which have probably come from some remote). Here we allow any location to have a priority and we will use that in a number of places. Here the thing I needed was to remove the "location" argument to `outpack_location_pull_packet` as it should not matter where we pull from.

This is part of a yak that I'm shaving - we need this in order to be able to automatically pull in missing packets which we'd need to be able to make sure that a location has all packets, which we need to have some policy on locations that they promise to be able to ship anything!

Issues for later:

* Can't easily view the current priorities of locations
* Can't edit the priority of a location

### CODE REVIEW

File: R/location.R [hunk 1]
```diff
@@ -264,3 +287,81 @@ location_pull_files_archive <- function(root, driver, id) {
...
+  ## In some cases we won't want local, make this easy to do:
+  if (!include_local) {
+    location <- setdiff(location, local)
+  }
+
+  location
+}
+
+
+location_pull_plan <- function(id, location, root) {
```
Comment (hillalex) [inline R/location.R]: So here "pull plan" is a noun, whereas in other methods "pull x" is an imperative? I know you don't like `get_plan` style function names but I find this a little confusing...

**Review by hillalex — COMMENTED** [review]
Just one comment about naming

File: R/location.R (hunk 1 above)
Comment (richfitz) [reply R/location.R]: how about build_plan?

**Review by hillalex — APPROVED** [review]

File: R/location.R [hunk 2]
```diff
@@ -15,13 +15,22 @@
 ##' @param path The path to the directory containing another outpack
 ##'   archive.
 ##'
+##' @param priority The priority of the location. This is used when
+##'   deciding where to pull packets from
+##'   ([outpack::outpack_location_pull_packet]), and will be used in
+##'   the query interface. A priority of 0 corresponds to the same
+##'   priority as local packets, while larger numbers have higher
+##'   priority and negative numbers have lower priority.  Ties will be
```
Comment (r-ash) [inline R/location.R:23]: ~~If we add a new location and don't specify priority then it has same priority as local. If we then try to pull a packet not specifying location there will be 50% chance it tries to pull from local and then errors?~~

~~Will we ever want to run `outpack_location_pull_packet` from local?~~

Nope - just seen the bit of code that excludes `local`

**Review by r-ash — APPROVED** [review]

**Pull request merged**